src/style/properties.js

```diff
@@ -576,6 +576,62 @@ export class DataDrivenProperty<T> implements Property<T, PossiblyEvaluatedPrope
     }
 }
 
+/**
+ * An implementation of `Property` for  data driven `line-pattern` which are transitioned by cross-fading
+ * rather than interpolation.
+ *
+ * @private
+ */
+
+export class CrossFadedDataDrivenProperty<T> extends DataDrivenProperty<?CrossFaded<T>> {
```

Comment:
per chat – we decided not to convert `CrossFadedDataDrivenProperty` to a top-level implementation of `Property` because it caused issues with `PossiblyEvaluatedPropertyValue` where the following wouldn't work

```es6
export class PossiblyEvaluatedPropertyValue<T> {
    property: DataDrivenProperty<T> | CrossFadedDataDrivenProperty<T>;
```

because we need the `<T>` in `property` to be `CrossFaded<T>` but the base `<T>` to be `string`